src/expression/embeddedDocs/function/matrix/diff.js

```diff
@@ -0,0 +1,18 @@
+export const diffDocs = {
+  name: 'diff',
+  category: 'Matrix',
+  syntax: [
+    'diff(arr)',
+    'diff(arr, dim)'
+  ],
+  description: [
+    'Create a new matrix or array with the difference of the passed matrix or array.',
+    'Dim parameter is optional and used to indicant the dimension of the array/matrix to apply the diffrence (if not used it is assumed as 0)'
+  ],
+  examples: [
+    'diff([1, 2, 4, 7, 0])',
+    'diff([1, 2, 4, 7, 0], 0)',
```

Comment:
Maybe this example should be a 2d array so you can actually see what the dimension does, like 

```
'diff([[1, 2], [3, 6]], 0)',
'diff([[1, 2], [3, 6]], 1)',
```

Reply:
Agreed, I think I did this because it was just a copy and paste but yes should definitely have more comprehensive examples